File: ems_hr_web/src/main/webapp/js/controller/rapController.js
 //控制层 
app.controller('rapController' ,function($scope,$controller   ,rapService){	
	
	$controller('baseeController',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
		rapService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){			
		rapService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	//查询实体 
	$scope.findOne=function(id){				
		rapService.findOne(id).success(
			function(response){
				$scope.rap= response;
			}
		);				
	}
	
	//保存 
	$scope.save=function(){				
		var object;//服务层对象
		if($scope.rap.id!=null){//如果有ID
			object=rapService.update( $scope.rap ); //修改
		}else{
			object=rapService.add( $scope.rap  );//增加
		}				
		object.success(
			function(response){
				if(response.success){
					//重新查询
                    alert(response.message);
		        	$scope.findAll();//重新加载
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	 
	//批量删除 
	$scope.dele=function(){

        if(confirm('您确定要删除吗？')) {
            if($scope.selectIds.length==0){
                alert("请选中要删除的数据！");
            }else {
                //获取选中的复选框
                rapService.dele( $scope.selectIds ).success(
                    function(response){
                        alert(response.message);
                        if(response.success){
                            $scope.findAll();//刷新列表,并清空数组
                            $scope.selectIds=[];
                        }
                    }
                );
            }

        }

	};
	
	$scope.searchEntity={};//定义搜索对象 
	
	//搜索
	$scope.search=function(page,rows){			
		rapService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
    
});	

File: ems_user_web/src/main/java/com/ems/user/controller/RapController.java
package com.ems.user.controller;
import java.util.List;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.ems.pojo.Rap;
import com.ems.service.RapService;

import entity.PageResult;
import entity.Result;
/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/rap")
public class RapController {

	@Reference
	private RapService rapService;
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<Rap> findAll(){
		return rapService.findAll();
	}
	
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult  findPage(int page,int rows){			
		return rapService.findPage(page, rows);
	}
	
	/**
	 * 增加
	 * @param rap
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody Rap rap){
		try {
			rapService.add(rap);
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}
	
	/**
	 * 修改
	 * @param rap
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody Rap rap){
		try {
			rapService.update(rap);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}	
	
	/**
	 * 获取实体
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public Rap findOne(Integer id){
		return rapService.findOne(id);		
	}
	
	/**
	 * 批量删除
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(Integer [] ids){
		try {
			rapService.delete(ids);
			return new Result(true, "删除成功"); 
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
	
		/**
	 * 查询+分页
	 * @param
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/search")
	public PageResult search(@RequestBody Rap rap, int page, int rows  ){
		return rapService.findPage(rap, page, rows);		
	}
	
}
